### PR TITLE
Add order in tutorial

### PR DESCRIPTION
**Pull request by makinzm**
## Motivation

Since the tutorial of Optuna does not describe how to proceed with the study, I felt that it is difficult for a beginner to get used to the system.

## Description of the changes

I just numbered the titles.

## Reference
[Optuna-V3-Roadmap](https://github.com/optuna/optuna/wiki/Optuna-V3-Roadmap)

### CODE REVIEW

File: tutorial/10_key_features/001_first.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """
 .. _first:
 
-Lightweight, versatile, and platform agnostic architecture
+1. Lightweight, versatile, and platform agnostic architecture
 ==========================================================
```
Comment (toshihikoyanase) [inline tutorial/10_key_features/001_first.py:5]: We got some warnings due to the formatting issue. The number of `=` in the next line of the title should be the same as the number of title characters. Add some `=` to it.
```
/home/runner/work/optuna/optuna/docs/source/tutorial/10_key_features/001_first.rst:24: WARNING: Title underline too short.

1. Lightweight, versatile, and platform agnostic architecture
==========================================================
/home/runner/work/optuna/optuna/docs/source/tutorial/10_key_features/002_configurations.rst:24: WARNING: Title underline too short.
looking for now-outdated files... none found

2. Pythonic Search Space
=====================
/home/runner/work/optuna/optuna/docs/source/tutorial/10_key_features/003_efficient_optimization_algorithms.rst:24: WARNING: Title underline too short.

3. Efficient Optimization Algorithms
=================================
/home/runner/work/optuna/optuna/docs/source/tutorial/10_key_features/004_distributed.rst:24: WARNING: Title underline too short.

4. Easy Parallelization
====================
/home/runner/work/optuna/optuna/docs/source/tutorial/10_key_features/005_visualization.rst:24: WARNING: Title underline too short.

5. Quick Visualization for Hyperparameter Optimization Analysis
============================================================
```

Comment (makinzm) [reply tutorial/10_key_features/001_first.py]: Thank you for your advice. I modified it. 

If you will check, I would appreciate it.

Comment (toshihikoyanase) [reply tutorial/10_key_features/001_first.py]: Thank you!

**Review by toshihikoyanase — APPROVED** [review]
LGTM. I believe that adding sequential numbers to the tutorial titles would be helpful for novice users.  Thank you!

The rendering result is as follows:
image

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**